proto-public/pbmesh/v2beta1/api_gateway.proto

```diff
@@ -52,3 +52,15 @@ message APIGatewayTLSConfiguration {
   // TLSParameters contains optional configuration for running TLS termination.
   optional pbproxystate.TLSParameters tls_parameters = 2;
 }
+
+
+// InlineCertificate specifies a certificate written directly into Consul's
+// store.
+message InlineCertificate {
+  option (hashicorp.consul.resource.spec) = {scope: SCOPE_NAMESPACE};
+
+  // PublicKey is the public key component of a certificate.
+  string PublicKey = 1;
+  // PrivateKey is the private key component of a certificate.
+  string PrivateKey = 2;
+}
```

Comment:
Thinking ahead a bit to other gateways and iterations on the API gateway, I think it'd be great to put this in its own file. Something like `gateway_certificates.proto`. We'll need similar pieces of configuration for terminating gateways, and it won't be specific to the API gateway at that point.